sound/soc/sof/intel/hda-ipc.c

```diff
@@ -172,18 +166,12 @@ irqreturn_t hda_dsp_ipc_irq_thread(int irq, void *context)
 		 */
 		spin_lock_irq(&sdev->ipc_lock);
 
-		/* handle immediate reply from DSP core - ignore ROM messages */
-		if (hda_dsp_ipc_is_sof(msg)) {
+		/* do not check reply if no message was initiated */
+		if (sdev->msg) {
```

Comment:
sorry, I remember @lyakh adding the call to hda_dsp_ipc_is_sof() to avoid handling some responses. I am not confident that your solution covers all the cases that were in that helper.